src/components/common/offline-banner/offline-banner.ts

```diff
@@ -10,4 +10,10 @@ export class OfflineBannerComponent {
   @Input()
   isOffline: boolean;
 
+  hasBeenOffline: boolean = false;
+
+  hasBeenOfflineSet() {
+    this.hasBeenOffline = true;
+  }
+
```

Comment:
Surely if `isOffline` is true we show the current banner i.e. `You are offline. Some features may be unavailable.`.

Therefore this isn't necessary as we would just use the reverse i.e. `!isOffline`

Reply:
Suggest using `ngOnChanges` to detect this like so

html
```
<div *ngIf="isOffline; else isOnline;" id="unauthenticated-mode-indicator">
  <label role="alert" id="unauth-text">You are offline. Some features may be unavailable.</label>
</div>
<ng-template #isOnline>
  <div *ngIf="hasBeenOffline" aria-label="Your device has re-connected" role="alert"></div>
</ng-template>

```

TS
```
import {
  Component, Input, OnChanges, SimpleChanges,
} from '@angular/core';

@Component({
  selector: 'offline-banner',
  templateUrl: 'offline-banner.html',
  styleUrls: ['offline-banner.scss'],
})
export class OfflineBannerComponent implements OnChanges {

  @Input()
  isOffline: boolean;

  hasBeenOffline: boolean = false;

  ngOnChanges(changes: SimpleChanges) {
    if (changes.isOffline?.currentValue === false) {
      this.hasBeenOffline = true;
    }
  }
}
```